### PR TITLE
Update ci builds versions

### PR DESCRIPTION
**Pull request by luismiramirez**
[skip changeset]

# Update CI build versions
CI was running with older versions of OTP and Elixir. Updated it to work
as the Elixir library.

# Remove compilation warnings
We are adding a compilation step on CI with warnings as errors. We have
to get rid of them to make the build pass.

### CODE REVIEW

File: .semaphore/semaphore.yml [hunk 1]
```diff
@@ -5,157 +5,93 @@ agent:
...
-          - elixir -v
-          - checkout
-          - mix local.rebar --force
-          - mix local.hex --force
-          - mix deps.get
-
-      jobs:
+        - name: mix compile --warnings-as-errors
+          commands:
+            - ERLANG_VERSION=24.0 ELIXIR_VERSION=1.12.2 . bin/setup
```
Comment (tombruijn) [inline .semaphore/semaphore.yml:23]:
```suggestion
            - ERLANG_VERSION=24.0 ELIXIR_VERSION=1.12.2 bin/setup
```

What's the dot for? I think we can remove it in all jobs

File: test/appsignal_phoenix/event_handler_test.exs [hunk 2]
```diff
@@ -34,42 +34,10 @@ defmodule Appsignal.Phoenix.EventHandlerTest do
...
-        conn: %Plug.Conn{},
-        log: :debug,
-        path_params: %{},
-        pipe_through: [:browser],
-        plug: AppsignalPhoenixExampleWeb.PageController,
-        plug_opts: plug_opts,
-        route: "/"
-      }
-    )
-  end
```
Comment (tombruijn) [inline test/appsignal_phoenix/event_handler_test.exs:71]: These functions were unused?

**Review by tombruijn — CHANGES_REQUESTED** [review]

File: .semaphore/semaphore.yml (hunk 1 above)
Comment (luismiramirez) [reply .semaphore/semaphore.yml:23]: I got it from the Elixir library

File: test/appsignal_phoenix/event_handler_test.exs (hunk 2 above)
Comment (luismiramirez) [reply test/appsignal_phoenix/event_handler_test.exs:71]: Yep, got a lot of warnings from these.

File: .semaphore/semaphore.yml (hunk 1 above)
Comment (tombruijn) [reply .semaphore/semaphore.yml:23]: it's probably not needed there either. Can you try without and if it works update the elixir package as well?

Comment (luismiramirez) [reply .semaphore/semaphore.yml]: Changes done.

**Review by tombruijn — APPROVED** [review]

**Pull request merged**